### ISSUE TITLE
The ClientConnection is closed and the ClientConnection is continually retried

### ISSUE THREAD

**Issue by yeyincai** [linkerd/linkerd#2266]
```java

com.twitter.finagle.ChannelWriteException: com.twitter.finagle.ChannelClosedException: null at remote address: 172.19.9.113/172.19.9.113:8080. Remote Info: Not Available from service: #/io.l5d.fs/foobar. Remote Info: Upstream Address: Not Available, Upstream id: Not Available, Downstream Address: /172.19.9.113:8080, Downstream label: #/io.l5d.fs/foobar, Trace Id: ef18b6ffecdb8ddf.179e11bba578422e<:ef18b6ffecdb8ddf
Caused by: com.twitter.finagle.ChannelClosedException: null at remote address: 172.19.9.113/172.19.9.113:8080. Remote Info: Not Available
	at com.twitter.finagle.ChannelException$.apply(Exceptions.scala:265)
	at com.twitter.finagle.netty4.transport.ChannelTransport$$anon$3.operationComplete(ChannelTransport.scala:104)
	at com.twitter.finagle.netty4.transport.ChannelTransport$$anon$3.operationComplete(ChannelTransport.scala:100)
	at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:511)
	at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:485)
	at io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:424)
	at io.netty.util.concurrent.DefaultPromise.addListener(DefaultPromise.java:162)
	at io.netty.channel.DefaultChannelPromise.addListener(DefaultChannelPromise.java:95)
	at io.netty.channel.DefaultChannelPromise.addListener(DefaultChannelPromise.java:30)
	at com.twitter.finagle.netty4.transport.ChannelTransport.write(ChannelTransport.scala:100)
	at com.twitter.finagle.netty4.http.Netty4ClientStreamTransport.write(StreamTransports.scala:171)
	at com.twitter.finagle.netty4.http.Netty4ClientStreamTransport.write(StreamTransports.scala:165)
	at com.twitter.finagle.http.HttpTransport.write(HttpTransport.scala:44)
	at com.twitter.finagle.http.HttpTransport.write(HttpTransport.scala:15)
	at com.twitter.finagle.http.codec.HttpClientDispatcher.dispatch(HttpClientDispatcher.scala:41)
	at com.twitter.finagle.http.codec.HttpClientDispatcher.dispatch(HttpClientDispatcher.scala:15)
	at com.twitter.finagle.dispatch.GenSerialClientDispatcher.tryDispatch(ClientDispatcher.scala:74)
	at com.twitter.finagle.dispatch.GenSerialClientDispatcher.$anonfun$apply$1(ClientDispatcher.scala:82)
	at com.twitter.finagle.dispatch.GenSerialClientDispatcher.$anonfun$apply$1$adapted(ClientDispatcher.scala:80)
	at com.twitter.util.ConstFuture$$anon$1.run(ConstFuture.scala:27)
	at com.twitter.concurrent.LocalScheduler$Activation.run(Scheduler.scala:198)
	at com.twitter.concurrent.LocalScheduler$Activation.submit(Scheduler.scala:157)
	at com.twitter.concurrent.LocalScheduler.submit(Scheduler.scala:274)
	at com.twitter.concurrent.Scheduler$.submit(Scheduler.scala:109)
	at com.twitter.util.Promise$WaitQueue.runInScheduler(Promise.scala:86)
	at com.twitter.util.Promise.updateIfEmpty(Promise.scala:776)
	at com.twitter.util.Promise.update(Promise.scala:748)
	at com.twitter.util.Promise.setException(Promise.scala:731)
	at com.twitter.concurrent.AsyncQueue.fail(AsyncQueue.scala:200)
	at com.twitter.finagle.netty4.transport.ChannelTransport.com$twitter$finagle$netty4$transport$ChannelTransport$$fail(ChannelTransport.scala:87)
	at com.twitter.finagle.netty4.transport.ChannelTransport$$anon$2.channelInactive(ChannelTransport.scala:179)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:245)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:231)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelInactive(AbstractChannelHandlerContext.java:224)
	at io.netty.channel.ChannelInboundHandlerAdapter.channelInactive(ChannelInboundHandlerAdapter.java:75)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:245)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:231)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelInactive(AbstractChannelHandlerContext.java:224)
	at io.netty.channel.ChannelInboundHandlerAdapter.channelInactive(ChannelInboundHandlerAdapter.java:75)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:245)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:231)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelInactive(AbstractChannelHandlerContext.java:224)
	at io.netty.channel.ChannelInboundHandlerAdapter.channelInactive(ChannelInboundHandlerAdapter.java:75)
	at io.netty.handler.codec.MessageAggregator.channelInactive(MessageAggregator.java:417)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:245)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:231)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelInactive(AbstractChannelHandlerContext.java:224)
	at io.netty.channel.CombinedChannelDuplexHandler$DelegatingChannelHandlerContext.fireChannelInactive(CombinedChannelDuplexHandler.java:420)
	at io.netty.handler.codec.ByteToMessageDecoder.channelInputClosed(ByteToMessageDecoder.java:390)
	at io.netty.handler.codec.ByteToMessageDecoder.channelInactive(ByteToMessageDecoder.java:355)
	at io.netty.handler.codec.http.HttpClientCodec$Decoder.channelInactive(HttpClientCodec.java:282)
	at io.netty.channel.CombinedChannelDuplexHandler.channelInactive(CombinedChannelDuplexHandler.java:223)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:245)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:231)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelInactive(AbstractChannelHandlerContext.java:224)
	at io.netty.channel.ChannelInboundHandlerAdapter.channelInactive(ChannelInboundHandlerAdapter.java:75)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:245)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:231)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelInactive(AbstractChannelHandlerContext.java:224)
	at io.netty.channel.ChannelInboundHandlerAdapter.channelInactive(ChannelInboundHandlerAdapter.java:75)
	at com.twitter.finagle.netty4.channel.ChannelRequestStatsHandler.channelInactive(ChannelRequestStatsHandler.scala:41)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:245)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:231)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelInactive(AbstractChannelHandlerContext.java:224)
	at io.netty.channel.ChannelInboundHandlerAdapter.channelInactive(ChannelInboundHandlerAdapter.java:75)
	at com.twitter.finagle.netty4.channel.ChannelStatsHandler.channelInactive(ChannelStatsHandler.scala:229)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:245)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:231)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelInactive(AbstractChannelHandlerContext.java:224)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelInactive(DefaultChannelPipeline.java:1429)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:245)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:231)
	at io.netty.channel.DefaultChannelPipeline.fireChannelInactive(DefaultChannelPipeline.java:947)
	at io.netty.channel.AbstractChannel$AbstractUnsafe$8.run(AbstractChannel.java:822)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute$$$capture(AbstractEventExecutor.java:163)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:404)
	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:335)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:897)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at com.twitter.finagle.util.BlockingTimeTrackingThreadFactory$$anon$1.run(BlockingTimeTrackingThreadFactory.scala:23)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.nio.channels.ClosedChannelException
	at io.netty.channel.AbstractChannel$AbstractUnsafe.write(...)(Unknown Source)
```

Our  uses Linkerd, but sometimes the above error occurs ， and the same request is repeated.
why?

**Comment by adleong** [comment]
Hi @yeyincai!  Can you describe in more detail the behavior that you're seeing?  What do you mean by the request is repeated?  

A ClosedChannelException generally means that the peer that Linkerd is connected to has closed the connection for some reason.

**Comment by yeyincai** [comment]
@adleong  Hi, This  ChannelWriteException  in to  RequeueFilter, and then repeat  execute .
```java

#see  the  com.twitter.finagle.service.RequeueFilter  code 
def apply(req: Req, service: Service[Req, Rep]): Future[Rep] = {
    retryBudget.deposit()
    val maxRetries = Math.ceil(maxRetriesPerReq * retryBudget.balance).toInt
    applyService(req, service, 0, maxRetries, retryBackoffs)
}

# this the maxRetries value  may be  =360
```

I don't know ，the   connected to has closed ,why   repeat  execute  send message

**Comment by adleong** [comment]
A requeued request will use an available connection from the connection pool or establish a new one if necessary.

**Comment by yeyincai** [comment]
A requeued request  ，when  connection pool  fails  ,too many retrys attempts will trigger  " FailureAccrualFactory marking connection to xxx-service as dead"  , this  is   correct ?

This linkerd  run log
`
I 0411 03:12:29.385 CST THREAD23 TraceId:c376e1dd9c31759f: FailureAccrualFactory marking connection to "xxxxxxxx" as dead. Remote Address: Inet(/10.1.1.1:1600,Map())

E 0411 03:12:29.656 CST THREAD23 TraceId:c376e1dd9c31759f: service failure: com.twitter.finagle.ChannelWriteException: com.twitter.finagle.ChannelClosedException: null at remote address: 10.1.1.1/10.1.1.1:1600. Remote Info: Not Available from service:xxxxxxxx. Remote Info: Upstream Address: Not Available, Upstream id: Not Available, Downstream Address: /10.1.1.1:1600, Downstream label: xxxxxxx, Trace Id: c376e1dd9c31759f.8d6e86c4dd8b9cc0<:c376e1dd9c31759f
`
